### PR TITLE
[VB Compiler Scanner ScanMultilineTrivia]

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
* Modify comment for clarity.
* Modify comparision slightly so it looks more like an inequality (or range check). 

Issue (#674)

### CODE REVIEW

**Comment by VSadov** [comment]
LGTM. Could you squash the commits?

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 1]
```diff
@@ -607,9 +607,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
             Dim ch = Peek()
 
-            ' optimization for a common case
-            ' the ASCII range between ': and ~ , with exception of except "'", "_" and R cannot start trivia
-            If ch > ":"c AndAlso ch <= "~"c AndAlso ch <> "'"c AndAlso ch <> "_"c AndAlso ch <> "R"c AndAlso ch <> "r"c Then
+            ' Optimization for a common case the ASCII range between ":"c (Exclusive) and "~"c (Incluse)
```
Comment (davkean) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: "Inclusive"

**Comment by davkean** [comment]
tag @dotnet/roslyn-compiler

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 1 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: Doh!, and will do once a test I running on a different branch is finished.

**Comment by AdamSpeight2008** [comment]
@davkean @VSadov  Finally Squashed.

**Comment by gafter** [comment]
While this is changing the same line that is discussed in issue #674, this change appears to just change the formatting of some code and does not address #674 (though I'm not sure that needs addressing). I don't understand why the parens were added to this code. They appear to be in an arbitrary place, but have no effect on the meaning of the code. That is quite confusing. I can't think of any reason the code base would be better with this change.

**Comment by AdamSpeight2008** [comment]
@gafter 
The comment had an ambiguity due it not indicating the inclusivity or exclusivity of the bounds.

Eg Is number between 0 and 9. There are four possible combinations 
```
0,1,2,3,4,5,6,7,8,9  (Inclusive, Inclusive) 
0,1,2,3,4,5,6,7,8    (inclusive, Exclusive)
  1,2,3,4,5,6,7,8,9  (Exclusive, Inclusive) 
  1,2,3,4,5,6,7,8    (Exclusive, Exclusive)
```
By stating them the ambiguity is removed.
Arranging the comparision as ` lower_bound < value <= upper_bound ` to me makes visually scanning the meaning easier. This is also help by the parens. 
The first set of parens match the `ASCII range between ":"c (Exclusive) and "~"c (Inclusive)` part and the other matches the excluded characters.

**Comment by gafter** [comment]
@AdamSpeight2008 Issue #674 suggests that a change to the code is needed. You just changed a comment and rearranged the code to mean the same thing it did before, but without addressing the change suggested in #674. Indeed, in some ways you made the comment worse. For example, "Optimization for a common case the ASCII range between" should have some punctuation between "case" and "the" - a newline served that purpose in the old code but now it seems to be a run-on sentence. I suggest a colon be added.

**Comment by AdamSpeight2008** [comment]
@gafter I did do something. Change the comment *(as well a slight change to code, to better reflect that comment)*, as per the discussion in the issue.

**Comment by gafter** [comment]
If you're changing the comment, then please (make it grammatical)[https://github.com/dotnet/roslyn/pull/6946#issuecomment-158693143].

**Comment by gafter** [comment]
Unless there is a compelling need for this change, I'd like to close this without accepting the change. While the comment itself was not crystal clear, the code *on the very next line* clarifies things precisely. The proposed new comment is not grammatical, and the proposed code change has a non-idiomatic pattern for comparisons (with the variable alternating on the left or right of the constant) and has an unknown performance impact.

**Pull request closed without merging**